plugins/package-managers/bazel/src/main/kotlin/Bazel.kt

```diff
@@ -306,6 +307,7 @@ class Bazel(
         packageIdsWithPathOverride.mapTo(result) {
             Package(
                 id = it,
+                purl = it.toPurl { it.lowercase() },
```

Comment:
Why did you decide to inject this into `toPurl` via a lambda vs. putting this into the `toPurl()` implementation directly?

Reply:
It simply didn't come to my mind 😄 But true: `toPurl()` knows the type and could already do any type-specific normalization in a central place, which is much better. I'd keep the argument for the low-level `createPurl()` function, though.

Reply:
Why keep it for `createPurl()`?

Reply:
With `createPurl()` being an `internal` function, I'd like to have some flexibility here, like being able to create invalid purls for testing or so. Also, it makes up for nicer code IMO if the normalization based on the type is only done in `toPurl()`.

Reply:
I've changed my mind as I've found a more elegant way to do this by defining the normalizations directly in the types.